linode/manager
# feat: [M3-7048] - Add DC-specific pricing for Kubernetes node pools

## Description 📝
This PR adds DC-specific pricing for Kubernetes node pools in the create flow and on the details page in add/resize drawers.

## Major Changes 🔄
**With the mocks and feature flag on:**
Kube Create page:
- When a region **is selected**, the Node Pools table displays the hourly and monthly pricing corresponding to the selected region.
- When a region **is not selected,** the Node Pools table does not display rows, and instead displays a message: "Select a Region to view plans and prices."
- The Cluster Summary displays correctly calculated DC-specific pricing based on the Linode's region – both Node Pool line items and the total.
- When either a region or a pool is not selected, the cluster summary does not display prices and displays helper text indicating what selections must be made for pricing and cluster creation.

Kube Details page:
- Kube cluster specs displays a correctly calculated total DC-specific total price of the cluster/month. 

Add a Node Pool drawer:
- The Node Pools table displays the hourly and monthly pricing corresponding to the region the cluster is in.
- The drawer copy displays correctly calculated DC-specific pricing based on the cluster's region.

Resize Pool drawer:
- The drawer copy displays correctly calculated DC-specific pricing based on the cluster's region.

## Preview 📷
| Flow/Component | Before  | After   |
| ------- | ------- | ------- |
| Create Cluster -> Node Pool Panel & Plan Selection | ![Screenshot 2023-08-30 at 4 14 48 PM](https://github.com/linode/manager/assets/114685994/157051b5-3a53-4e25-b6ee-8bbf2dda90ff) ![Screenshot 2023-08-30 at 4 12 25 PM](https://github.com/linode/manager/assets/114685994/45a9018f-75ae-4c56-a301-3ccffdad10fb) | ![Screenshot 2023-08-30 at 10 35 21 AM](https://github.com/linode/manager/assets/114685994/77526dd4-ae32-44c4-9989-39c4b90b2dde) ![Screenshot 2023-08-30 at 3 20 42 PM](https://github.com/linode/manager/assets/114685994/7c9ef500-07f9-4903-b73b-676aad97f444)|
| Create Cluster -> Checkout Bar | ![Screenshot 2023-08-30 at 4 13 05 PM](https://github.com/linode/manager/assets/114685994/febb4136-6740-4119-b5e7-29b5f9e94d0d) | ![Screenshot 2023-08-30 at 4 21 03 PM](https://github.com/linode/manager/assets/114685994/bad5aab3-473d-435b-b407-a939903e8767)|
| Kubenetes Cluster Details -> Specs | ![Screenshot 2023-08-30 at 3 30 22 PM](https://github.com/linode/manager/assets/114685994/8cc6fda6-e6d5-4706-b65b-8c3c44f695be) | ![Screenshot 2023-08-30 at 3 30 14 PM](https://github.com/linode/manager/assets/114685994/eeb54d28-5be2-434f-a826-1d483ee8cb2e)|
| Kubernetes Cluster Details -> Add Node Pool Drawer  | ![Screenshot 2023-08-30 at 3 30 46 PM](https://github.com/linode/manager/assets/114685994/418cd599-063f-4769-9a92-cafc470fb976) | ![Screenshot 2023-08-30 at 3 31 02 PM](https://github.com/linode/manager/assets/114685994/e99676cf-6bc7-4f2e-86da-9dafe1e84148) |
| Kubernetes Cluster Details -> Resize Node Pool Drawer  | ![Screenshot 2023-08-30 at 3 31 43 PM](https://github.com/linode/manager/assets/114685994/f4eae87c-5803-4ca9-9ec3-3af748b34208) | ![Screenshot 2023-08-30 at 3 31 27 PM](https://github.com/linode/manager/assets/114685994/6db5ba55-c457-4125-b108-6716d1696857)|

## How to test 🧪
1. **How to setup test environment?**
- Ensure `DC-Specific Pricing` flag and MSW are both **on**.
- To test, I temporarily modified the `kubernetesAPIResponse` and `kubernetesClusterFactory` factories and replaces `region: 'us-central'` with `region: 'id-cgk'`.

3. **How to verify changes?**

- Test that the above changes are visible with the DC-Specific Pricing flag **on** and not visible with the flag **off**.
- Confirm that the plans selection cards in mobile view update dynamically with region selection when the flag is on and the notice looks good.

5. **How to run Unit or E2E tests?**
```
yarn test NodePoolPanel kubeUtils ResizeNodePoolDrawer KubeCheckoutBar KubernetesPlanContainer KubernetesPlanSelection
```

I did not cover the Add/Resize Node Pool drawers. There is an `lke-update.spec.ts` testing adding and resizing node pools; these should be updated in an e2e ticket.
